src/main/kotlin/org/rust/ide/refactoring/extractEnumVariant/RsExtractEnumVariantProcessor.kt

```diff
@@ -0,0 +1,76 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.refactoring.extractEnumVariant
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.search.searches.ReferencesSearch
+import com.intellij.refactoring.BaseRefactoringProcessor
+import com.intellij.usageView.BaseUsageViewDescriptor
+import com.intellij.usageView.UsageInfo
+import com.intellij.usageView.UsageViewDescriptor
+import org.rust.ide.inspections.import.RsImportHelper
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.RsReferenceElement
+import org.rust.lang.core.psi.ext.descendantOfTypeStrict
+import org.rust.lang.core.psi.ext.parentEnum
+
+class RsExtractEnumVariantProcessor(project: Project,
+                                    private val editor: Editor,
+                                    private val ctx: RsEnumVariant) : BaseRefactoringProcessor(project) {
+    override fun findUsages(): Array<UsageInfo> {
+        return ReferencesSearch.search(ctx).map { UsageInfo(it) }.toTypedArray()
+    }
+
+    override fun performRefactoring(usages: Array<out UsageInfo>) {
+        val project = ctx.project
+        val factory = RsPsiFactory(project)
+        val element = createElement(ctx, factory)
+        val references = ReferencesSearch.search(ctx)
```

Comment:
One of the main purpose of `BaseRefactoringProcessor` is split reading phase (usage searching) from writing phase (usage updating). Reading phase is already done here via invocation of `findUsages` method in `BaseRefactoringProcessor` implementation and the result is passed as `usages`.
So you don't need to search them again. Just use `usages `

Reply:
Oh, sorry, this was actually a copy paste error. Thanks.